### PR TITLE
[ASM] adding missing usings to dispose

### PR DESCRIPTION
**Pull request by anna-git**

### CODE REVIEW

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #3622 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8): 
* All benchmarks have **the same** speed
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`WriteAndFlushEnrichedTraces`|net472|777μs|702ns|2.63μs|0.386| 0|0|3.22&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|561μs|980ns|3.8μs|0| 0|0|2.63&nbsp;KB|
|#3622|`WriteAndFlushEnrichedTraces`|net472|755μs|616ns|2.22μs|0.379| 0|0|3.22&nbsp;KB|
|#3622|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|556μs|405ns|1.57μs|0| 0|0|2.63&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AppSecBodyBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`AllCycleSimpleBody`|net472|22.8μs|11ns|42.5ns|0.276| 0|0|1.78&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`AllCycleSimpleBody`|netcoreapp3.1|24μs|134ns|890ns|0.0124| 0|0|1.65&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`AllCycleMoreComplexBody`|net472|190μs|58ns|217ns|1.99| 0|0|13.03&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`AllCycleMoreComplexBody`|netcoreapp3.1|179μs|710ns|2.75μs|0.0904| 0|0|12.11&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`BodyExtractorSimpleBody`|net472|285ns|0.201ns|0.779ns|0.0573| 0|0|361&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`BodyExtractorSimpleBody`|netcoreapp3.1|236ns|0.0854ns|0.319ns|0.00372| 0|0|272&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`BodyExtractorMoreComplexBody`|net472|15.9μs|43.5ns|163ns|1.21| 0.0159|0|7.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`BodyExtractorMoreComplexBody`|netcoreapp3.1|12.9μs|5.79ns|21.7ns|0.0899| 0|0|6.75&nbsp;KB|
|#3622|`AllCycleSimpleBody`|net472|25.1μs|20.7ns|80.1ns|0.275| 0|0|1.78&nbsp;KB|
|#3622|`AllCycleSimpleBody`|netcoreapp3.1|22.8μs|115ns|516ns|0.0115| 0|0|1.65&nbsp;KB|
|#3622|`AllCycleMoreComplexBody`|net472|193μs|96.8ns|375ns|2.03| 0|0|13.03&nbsp;KB|
|#3622|`AllCycleMoreComplexBody`|netcoreapp3.1|182μs|816ns|3.16μs|0.0915| 0|0|12.11&nbsp;KB|
|#3622|`BodyExtractorSimpleBody`|net472|298ns|0.394ns|1.47ns|0.0573| 0|0|361&nbsp;B|
|#3622|`BodyExtractorSimpleBody`|netcoreapp3.1|240ns|0.123ns|0.478ns|0.00374| 0|0|272&nbsp;B|
|#3622|`BodyExtractorMoreComplexBody`|net472|15.6μs|14ns|54.1ns|1.21| 0.0156|0|7.62&nbsp;KB|
|#3622|`BodyExtractorMoreComplexBody`|netcoreapp3.1|12.9μs|4.7ns|17ns|0.0906| 0|0|6.75&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Unknown :shrug: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`SendRequest`|netcoreapp3.1|279μs|1.58μs|11μs|0.139| 0|0|20.96&nbsp;KB|
|#3622|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#3622|`SendRequest`|netcoreapp3.1|265μs|1.44μs|8.14μs|0.268| 0|0|20.96&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`ExecuteNonQuery`|net472|1.84μs|2.39ns|9.27ns|0.163| 0.000919|0|1.03&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`ExecuteNonQuery`|netcoreapp3.1|1.48μs|2.98ns|11.1ns|0.014| 0|0|1.02&nbsp;KB|
|#3622|`ExecuteNonQuery`|net472|1.94μs|2.47ns|9.55ns|0.163| 0.000962|0|1.03&nbsp;KB|
|#3622|`ExecuteNonQuery`|netcoreapp3.1|1.43μs|1.97ns|7.63ns|0.0133| 0|0|1.02&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`CallElasticsearch`|net472|2.41μs|1.03ns|3.99ns|0.196| 0|0|1.24&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`CallElasticsearch`|netcoreapp3.1|1.54μs|1.32ns|5.1ns|0.0161| 0|0|1.18&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`CallElasticsearchAsync`|net472|2.63μs|1.87ns|7.26ns|0.218| 0|0|1.37&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`CallElasticsearchAsync`|netcoreapp3.1|1.61μs|0.851ns|2.95ns|0.0176| 0|0|1.3&nbsp;KB|
|#3622|`CallElasticsearch`|net472|2.51μs|0.7ns|2.62ns|0.196| 0|0|1.24&nbsp;KB|
|#3622|`CallElasticsearch`|netcoreapp3.1|1.58μs|1.6ns|5.98ns|0.016| 0|0|1.18&nbsp;KB|
|#3622|`CallElasticsearchAsync`|net472|2.63μs|1.58ns|6.11ns|0.218| 0|0|1.37&nbsp;KB|
|#3622|`CallElasticsearchAsync`|netcoreapp3.1|1.61μs|0.619ns|2.4ns|0.0177| 0|0|1.3&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`ExecuteAsync`|net472|2.61μs|0.774ns|3ns|0.237| 0|0|1.5&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`ExecuteAsync`|netcoreapp3.1|1.8μs|0.5ns|1.8ns|0.0189| 0|0|1.42&nbsp;KB|
|#3622|`ExecuteAsync`|net472|2.62μs|0.802ns|3.11ns|0.238| 0|0|1.5&nbsp;KB|
|#3622|`ExecuteAsync`|netcoreapp3.1|1.77μs|0.513ns|1.92ns|0.0194| 0|0|1.42&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`SendAsync`|net472|7.36μs|2.96ns|11.5ns|0.56| 0|0|3.54&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`SendAsync`|netcoreapp3.1|4.69μs|2.38ns|8.92ns|0.0445| 0|0|3.21&nbsp;KB|
|#3622|`SendAsync`|net472|7.33μs|2.66ns|9.96ns|0.562| 0|0|3.54&nbsp;KB|
|#3622|`SendAsync`|netcoreapp3.1|4.69μs|2.26ns|8.16ns|0.0421| 0|0|3.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`EnrichedLog`|net472|2.83μs|1.66ns|6.22ns|0.301| 0|0|1.9&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`EnrichedLog`|netcoreapp3.1|2.35μs|1.62ns|5.84ns|0.0258| 0|0|1.93&nbsp;KB|
|#3622|`EnrichedLog`|net472|2.75μs|3.76ns|14.5ns|0.301| 0|0|1.9&nbsp;KB|
|#3622|`EnrichedLog`|netcoreapp3.1|2.22μs|0.619ns|2.4ns|0.0266| 0|0|1.93&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`EnrichedLog`|net472|148μs|124ns|480ns|0.736| 0.221|0|4.74&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`EnrichedLog`|netcoreapp3.1|116μs|119ns|445ns|0.0586| 0|0|4.58&nbsp;KB|
|#3622|`EnrichedLog`|net472|147μs|106ns|398ns|0.736| 0.221|0|4.74&nbsp;KB|
|#3622|`EnrichedLog`|netcoreapp3.1|118μs|126ns|487ns|0.0588| 0|0|4.58&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`EnrichedLog`|net472|5.76μs|2.48ns|9.62ns|0.583| 0.00287|0|3.68&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`EnrichedLog`|netcoreapp3.1|4.28μs|3.45ns|12.9ns|0.0551| 0|0|4&nbsp;KB|
|#3622|`EnrichedLog`|net472|5.51μs|3.17ns|12.3ns|0.583| 0.00274|0|3.68&nbsp;KB|
|#3622|`EnrichedLog`|netcoreapp3.1|4.3μs|1.5ns|5.63ns|0.0536| 0|0|4&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`SendReceive`|net472|2.24μs|1.19ns|4.31ns|0.231| 0|0|1.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`SendReceive`|netcoreapp3.1|1.78μs|0.591ns|2.21ns|0.0186| 0|0|1.4&nbsp;KB|
|#3622|`SendReceive`|net472|2.3μs|2.33ns|9.04ns|0.23| 0|0|1.45&nbsp;KB|
|#3622|`SendReceive`|netcoreapp3.1|1.81μs|0.784ns|3.04ns|0.019| 0|0|1.4&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`EnrichedLog`|net472|4.86μs|1.06ns|3.98ns|0.367| 0|0|2.32&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`EnrichedLog`|netcoreapp3.1|3.99μs|1.31ns|4.91ns|0.0259| 0|0|1.89&nbsp;KB|
|#3622|`EnrichedLog`|net472|4.75μs|2.08ns|7.77ns|0.368| 0|0|2.32&nbsp;KB|
|#3622|`EnrichedLog`|netcoreapp3.1|4.08μs|0.995ns|3.72ns|0.0245| 0|0|1.89&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`StartFinishSpan`|net472|1.1μs|0.245ns|0.918ns|0.141| 0|0|891&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`StartFinishSpan`|netcoreapp3.1|942ns|0.434ns|1.62ns|0.0114| 0|0|840&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`StartFinishScope`|net472|1.37μs|0.43ns|1.66ns|0.154| 0|0|971&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`StartFinishScope`|netcoreapp3.1|1.06μs|0.399ns|1.44ns|0.0133| 0|0|960&nbsp;B|
|#3622|`StartFinishSpan`|net472|1.08μs|0.326ns|1.26ns|0.141| 0|0|891&nbsp;B|
|#3622|`StartFinishSpan`|netcoreapp3.1|930ns|0.311ns|1.12ns|0.0114| 0|0|840&nbsp;B|
|#3622|`StartFinishScope`|net472|1.4μs|0.399ns|1.44ns|0.154| 0|0|971&nbsp;B|
|#3622|`StartFinishScope`|netcoreapp3.1|1.06μs|0.413ns|1.6ns|0.0131| 0|0|960&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`RunOnMethodBegin`|net472|1.48μs|0.327ns|1.22ns|0.154| 0|0|971&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/340b860420e2a5ad38d0698e2785af340b0556c8)|`RunOnMethodBegin`|netcoreapp3.1|1.11μs|0.83ns|2.99ns|0.0128| 0|0|960&nbsp;B|
|#3622|`RunOnMethodBegin`|net472|1.44μs|0.909ns|3.4ns|0.154| 0|0|971&nbsp;B|
|#3622|`RunOnMethodBegin`|netcoreapp3.1|1.11μs|0.389ns|1.51ns|0.0134| 0|0|960&nbsp;B|
</details>

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #3622 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/67cc0d303d3b137a0250d7e5384a94157c0a8c02) will **not change** line coverage
:heavy_check_mark: Merging #3622 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/67cc0d303d3b137a0250d7e5384a94157c0a8c02) will **not change** branch coverage
:no_entry: Merging #3622 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/67cc0d303d3b137a0250d7e5384a94157c0a8c02) will will **increase** complexity by `13`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/67cc0d303d3b137a0250d7e5384a94157c0a8c02) | #3622       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `20926` / `29470` | `20982` / `29481` |          |
| Lines %   | `71%`      | `71%`      |  `0%` :heavy_check_mark:  |
| Branches  | `12543` / `18985` | `12587` / `19001` |          |
| Branches %| `66%`      | `66%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `20482`      | `20495`        |  `13`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=123590&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/123590/artifacts?artifactName=Code%20Coverage%20Report_123590&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3622](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=123860&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/123860/artifacts?artifactName=Code%20Coverage%20Report_123860&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/67cc0d303d3b137a0250d7e5384a94157c0a8c02) | #3622       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `71%`      | `71%`       |  `0%` :heavy_check_mark:  |
| Branches %| `66%`      | `66%`       |  `0%` :heavy_check_mark:  |
| Complexity| `20482`      | `20495`       |  `13` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.ClusterNodeIntegrationTer](https://github.com/DataDog/dd-trace-dotnet/tree/b16ce76d5be45a03b198c2a50fa96a81faade2a7/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/ClusterNodeIntegrationTer.cs) | `0%` :heavy_check_mark: | `-100%` :no_entry:   | `1` :no_entry: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.ClusterNodeIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/b16ce76d5be45a03b198c2a50fa96a81faade2a7/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/ClusterNodeIntegrationBis.cs) | `0%` :heavy_check_mark: | `-50%` :no_entry:   | `1` :no_entry: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.ClusterNodeIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/b16ce76d5be45a03b198c2a50fa96a81faade2a7/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/ClusterNodeIntegration.cs) | `0%` :heavy_check_mark: | `-17%` :no_entry:   | `11` :no_entry: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=123590&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/123590/artifacts?artifactName=Code%20Coverage%20Report_123590&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3622](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=123860&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/123860/artifacts?artifactName=Code%20Coverage%20Report_123860&api-version=6.0&%24format=zip)

File: tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs [hunk 1]
```diff
@@ -122,7 +122,7 @@ public bool ToggleRules(IDictionary<string, bool> ruleStatus)
                 return false;
             }
 
-            var encoded = encoder.Encode(ruleStatus);
+            using var encoded = encoder.Encode(ruleStatus);
```
Comment (GreenMatan) [inline tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs:125]: Maybe you want to also include `using` here:
https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/AppSec/Waf/Initialization/WafConfigurator.cs#L175

**Review by GreenMatan — APPROVED** [review]

File: tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs (hunk 1 above)
Comment (anna-git) [reply tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs:125]: this one we return it and need it for a later stage to configure the waf [here ](https://github.com/DataDog/dd-trace-dotnet/blob/a8abc21d58ea9d91bc0b5b2ba5f9d265f97e41d9/tracer/src/Datadog.Trace/AppSec/Waf/Initialization/WafConfigurator.cs#L110)

**Review by robertpi — APPROVED** [review]

**Pull request merged**